### ISSUE TITLE
Incorrect use of Redstone oracle exposes protocol to incorrect collateral evaluation 

### ISSUE THREAD

**Issue by abhishekvispute** [raft-fi/contracts#317]
### Severity
High

### Description
As per discussion with the raft team, they are looking to use Tellor or Redstone as their secondary oracle.
If the circuit breaks in the case of the primary oracle(chainlink), the protocol uses the secondary oracle to fetch prices and function.</br>
In both chainlink and tellor implementation, `_convertIntoWstETHPrice()` is used to convert the stETH price into wstETH price, the two operations done here are scaling decimals to 18 and converting it into wstETH using `wstETH.stEthPerToken()`.
```
BasePriceOracle.sol

  function _convertIntoWstETHPrice(uint256 price, uint256 answerDigits) internal view returns (uint256) {
        return _scalePriceByDigits(price, answerDigits).mulDown(wstETH.stEthPerToken());
    }
   function _scalePriceByDigits(uint256 price, uint256 answerDigits) internal pure returns (uint256) {
        /*
        * Convert the price returned by the oracle to an 18-digit decimal for use by Raft.
        */
        if (answerDigits > TARGET_DIGITS) {
            // Scale the returned price value down to Raft's target precision
            return price / (10 ** (answerDigits - TARGET_DIGITS));
        }
        if (answerDigits < TARGET_DIGITS) {
            // Scale the returned price value up to Raft's target precision
            return price * (10 ** (TARGET_DIGITS - answerDigits));
        }
        return price;
    }
``` 

**However, this is missed in the case of the RedStone oracle.**

https://github.com/tempusfinance/raft-contracts/blob/7c98a5141e308f143a9fc45b4a3c33b43df489fd/contracts/Oracles/RedstoneConsumerIntegration.sol#L11

`RedstoneConsumerIntegration.sol` defines `STETH` as `DATA_FEED_ID` as constant, and hence use STETH price feed only.
However, the conversion for _convertIntoWstETHPrice is not done in ``RedstonePriceOracle.sol`` to accommodate STETH price feed.

https://github.com/tempusfinance/raft-contracts/blob/7c98a5141e308f143a9fc45b4a3c33b43df489fd/contracts/Oracles/RedstonePriceOracle.sol#L26-L32

Hence the price of stETH would be used instead of wstETH to evaluate wstETH collateral value, exposing protocol to incorrect evaluation.

### Attack scenario
1. Raft's primary oracle fails.
2. Raft start using their secondary oracle : RedStone
3. Raft underestimates the collateral value 
4. Resulting in unfair liquidations

The underestimation happens since wstETH is generally valued more than stETH
https://www.coingecko.com/en/coins/wrapped-steth
https://www.coingecko.com/en/coins/lido-staked-ether

### Recommendation
- Consider using wstETH as a data feed in place of stETH, hence not needing any extra conversions
https://app.redstone.finance/#/app/token/WSTETH
- Consider doing the appropriate conversions if you are using stETH as a data feed
https://app.redstone.finance/#/app/token/STETH

**Comment by abhishekvispute** [comment]
@mijovic having a failing test PR of this issue may not work, since I can not replicate the circumstances locally, at best can do mocked setup, but that's not a good representation of the issue from any angle.

Let me know if you would still like failing test 
I can write a test on forked setup, but for that, I will need the address of `redstoneConsumerBase`, I couldn't find so on Redstone docs, let me know if you know of any

**Comment by StefanVitor** [comment]
Fixed by #386
